Email settings require using encryption and usernames

It seems I am required to use authentication when trying to use email.

I self-host a mailrelay, which handles talking to my "real" email-provider, which allows me to send email, even when not connected - as the email is cached there before sending.

However, that relay has no login, and that doesn't work.


````
[NLP]: [2026-02-27] :: [15:03:23] - [ERROR] [EMAIL CONFIG] Required field not set for 'Email': SMTP Username
[NLP]: [2026-02-27] :: [15:03:23] - [ERROR] [EMAIL CONFIG] Required field not set for 'Email': SMTP Password
````

I did configure some dummy username/password, and that lead to a new issue - it apparently also doesn't support running without STARTTLS
````
[NLP]: [2026-02-27] :: [15:09:58] - [ERROR] An error has occured: System.NotSupportedException: The SMTP server does not support the STARTTLS extension.
````

--- a/Jellyfin.Plugin.Newsletters/Clients/Email/smtp.cs
+++ b/Jellyfin.Plugin.Newsletters/Clients/Email/smtp.cs
@@ -63,7 +63,7 @@ public class SmtpMailer(IServerApplicationHost appHost,
             Logger.Debug($"Sending out test mail for '{emailConfig.Name}'!");
             string smtpAddress = emailConfig.SMTPServer;
             int portNumber = emailConfig.SMTPPort;
-            bool enableSSL = true;
+            bool enableSSL = emailConfig.EnableSsl;
             string emailFromAddress = emailConfig.FromAddr;
             string username = emailConfig.SMTPUser;
             string password = emailConfig.SMTPPass;
@@ -223,7 +223,7 @@ public class SmtpMailer(IServerApplicationHost appHost,
             Logger.Debug($"Sending out mail for '{emailConfig.Name}'!");
             string smtpAddress = emailConfig.SMTPServer;
             int portNumber = emailConfig.SMTPPort;
-            bool enableSSL = true;
+            bool enableSSL = emailConfig.EnableSsl;
             string emailFromAddress = emailConfig.FromAddr;
             string username = emailConfig.SMTPUser;
             string password = emailConfig.SMTPPass;

--- a/Jellyfin.Plugin.Newsletters/Configuration/EmailConfiguration.cs
+++ b/Jellyfin.Plugin.Newsletters/Configuration/EmailConfiguration.cs
@@ -102,4 +102,9 @@ public class EmailConfiguration : INewsletterConfiguration
     /// Gets or sets a value indicating whether to include upcoming items in the newsletter.
     /// </summary>
     public bool NewsletterOnUpcomingItemEnabled { get; set; } = false;
+
+    /// <summary>
+    /// Gets or sets a value indicating whether SSL is enabled for the SMTP connection.
+    /// </summary>
+    public bool EnableSsl { get; set; } = true;
 }

--- a/Jellyfin.Plugin.Newsletters/Configuration/configPage.html
+++ b/Jellyfin.Plugin.Newsletters/Configuration/configPage.html
@@ -748,10 +748,10 @@
                 config.MoviesUpdateEmbedColor = document.getElementById('discord-movies-update-color-' + id).value;
 
                 // Update Event Toggles
-                if (document.getElementById('discord-event-add-' + id)) config.NewsletterOnItemAddedEnabled = document.getElementById('discord-event-add-' + id).checked;
-                if (document.getElementById('discord-event-update-' + id)) config.NewsletterOnItemUpdatedEnabled = document.getElementById('discord-event-update-' + id).checked;
-                if (document.getElementById('discord-event-delete-' + id)) config.NewsletterOnItemDeletedEnabled = document.getElementById('discord-event-delete-' + id).checked;
-                if (document.getElementById('discord-event-upcoming-' + id)) config.NewsletterOnUpcomingItemEnabled = document.getElementById('discord-event-upcoming-' + id).checked;
+                config.NewsletterOnItemAddedEnabled = document.getElementById('discord-event-add-' + id).checked;
+                config.NewsletterOnItemUpdatedEnabled = document.getElementById('discord-event-update-' + id).checked;
+                config.NewsletterOnItemDeletedEnabled = document.getElementById('discord-event-delete-' + id).checked;
+                config.NewsletterOnUpcomingItemEnabled = document.getElementById('discord-event-upcoming-' + id).checked;
 
                 // Update Library Selections
                 var seriesChecked = document.querySelectorAll('.discord-series-lib-' + id + ':checked');
@@ -902,10 +902,10 @@
                 config.EpisodesEnabled = document.getElementById('telegram-episodes-' + id).checked;
 
                 // Update Event Toggles
-                if (document.getElementById('telegram-event-add-' + id)) config.NewsletterOnItemAddedEnabled = document.getElementById('telegram-event-add-' + id).checked;
-                if (document.getElementById('telegram-event-update-' + id)) config.NewsletterOnItemUpdatedEnabled = document.getElementById('telegram-event-update-' + id).checked;
-                if (document.getElementById('telegram-event-delete-' + id)) config.NewsletterOnItemDeletedEnabled = document.getElementById('telegram-event-delete-' + id).checked;
-                if (document.getElementById('telegram-event-upcoming-' + id)) config.NewsletterOnUpcomingItemEnabled = document.getElementById('telegram-event-upcoming-' + id).checked;
+                config.NewsletterOnItemAddedEnabled = document.getElementById('telegram-event-add-' + id).checked;
+                config.NewsletterOnItemUpdatedEnabled = document.getElementById('telegram-event-update-' + id).checked;
+                config.NewsletterOnItemDeletedEnabled = document.getElementById('telegram-event-delete-' + id).checked;
+                config.NewsletterOnUpcomingItemEnabled = document.getElementById('telegram-event-upcoming-' + id).checked;
 
                 // Update Library Selections
                 var seriesChecked = document.querySelectorAll('.telegram-series-lib-' + id + ':checked');
@@ -995,7 +995,8 @@
                     NewsletterOnItemAddedEnabled: true,
                     NewsletterOnItemUpdatedEnabled: false,
                     NewsletterOnItemDeletedEnabled: true,
-                    NewsletterOnUpcomingItemEnabled: false
+                    NewsletterOnUpcomingItemEnabled: false,
+                    EnableSsl: true
                 };
                 emailConfigs.push(newConfig);
                 renderEmailConfigs();
@@ -1043,12 +1044,13 @@
                 config.Body = document.getElementById('email-body-' + id).value;
                 config.Entry = document.getElementById('email-entry-' + id).value;
                 config.TemplateCategory = document.getElementById('email-category-' + id).value;
+                config.EnableSsl = document.getElementById('email-ssl-' + id).checked;
 
                 // Update Event Toggles
-                if (document.getElementById('email-event-add-' + id)) config.NewsletterOnItemAddedEnabled = document.getElementById('email-event-add-' + id).checked;
-                if (document.getElementById('email-event-update-' + id)) config.NewsletterOnItemUpdatedEnabled = document.getElementById('email-event-update-' + id).checked;
-                if (document.getElementById('email-event-delete-' + id)) config.NewsletterOnItemDeletedEnabled = document.getElementById('email-event-delete-' + id).checked;
-                if (document.getElementById('email-event-upcoming-' + id)) config.NewsletterOnUpcomingItemEnabled = document.getElementById('email-event-upcoming-' + id).checked;
+                config.NewsletterOnItemAddedEnabled = document.getElementById('email-event-add-' + id).checked;
+                config.NewsletterOnItemUpdatedEnabled = document.getElementById('email-event-update-' + id).checked;
+                config.NewsletterOnItemDeletedEnabled = document.getElementById('email-event-delete-' + id).checked;
+                config.NewsletterOnUpcomingItemEnabled = document.getElementById('email-event-upcoming-' + id).checked;
 
                 // Update Library Selections
                 var seriesChecked = document.querySelectorAll('.email-series-lib-' + id + ':checked');
@@ -1105,6 +1107,12 @@
                     html += '        <label class="inputLabel inputLabelUnfocused">SMTP Password:</label>';
                     html += '        <input id="email-pass-' + config.Id + '" type="password" is="emby-input" value="' + (config.SMTPPass || '') + '" onchange="updateEmailConfigFromUI(\'' + config.Id + '\')" />';
                     html += '      </div>';
+                    html += '    </div>';
+                    html += '    <div class="checkboxContainer">';
+                    html += '      <label class="emby-checkbox-label">';
+                    html += '        <input id="email-ssl-' + config.Id + '" type="checkbox" is="emby-checkbox" ' + (config.EnableSsl !== false ? 'checked' : '') + ' onchange="updateEmailConfigFromUI(\'' + config.Id + '\')" />';
+                    html += '        <span>Enable SSL/TLS</span>';
+                    html += '      </label>';
                     html += '    </div>';
                     html += '    <div class="inputContainer">';
                     html += '      <label class="inputLabel inputLabelUnfocused">To Address:</label>';

